armi/physics/neutronics/crossSectionSettings.py

```diff
@@ -125,65 +181,136 @@ def isPregenerated(self):
+            for var, val in self:
+                if var not in validOptions and val is not None:
+                    invalids.append((var, val))
+
+        if invalids:
+            runLog.warning(
+                f"The following inputs in {self} are not valid when `{self.geometry}` geometry type is set:"
+            )
+            for var, val in invalids:
+                runLog.warning(f"\tAttribute: {var}, Value: {val}")
+            runLog.warning(
+                f"The valid options for the `{self.geometry}` geometry are: {validOptions}"
+            )
+
     def setDefaults(self, blockRepresentation, validBlockTypes):
-        # TODO: explain this logic
+        """
+        This sets the defaults based on some recommended values based on the geometry type.
+
+        Notes
+        ----
+        This is useful in cases where the user wants the cross sections to be generated for
+        the defined cross section types in the model, but doesn't have to explictly set
+        all the options.
+
+        The supported defaults for the geometry are: ["0D", "1D slab", "1D cylinder", "2D hex"].
+        """
         validBlockTypes = None if validBlockTypes else ["fuel"]
-        if self.geometry == "0D":
-            defaults = dict(
-                criticalBuckling=True,
-                numExternalRings=None,
-                blockRepresentation=blockRepresentation,
-                validBlockTypes=validBlockTypes,
-            )
+
+        if self.isPregenerated:
+            defaults = {
+                CONF_FILE_LOCATION: self.fileLocation,
+                CONF_BLOCK_REPRESENTATION: blockRepresentation,
+            }
+
+        elif self.geometry == "0D":
+            defaults = {
+                CONF_GEOM: "0D",
+                CONF_BUCKLING: True,
+                CONF_DRIVER: "",
+                CONF_BLOCK_REPRESENTATION: blockRepresentation,
+                CONF_BLOCKTYPES: validBlockTypes,
+            }
         elif self.geometry == "1D slab":
-            defaults = dict(
-                criticalBuckling=False,
-                mergeIntoClad=[],
-                blockRepresentation=blockRepresentation,
-                validBlockTypes=validBlockTypes,
-            )
+            defaults = {
+                CONF_GEOM: "1D slab",
+                CONF_MESH_PER_CM: 1.0,
+                CONF_BLOCK_REPRESENTATION: blockRepresentation,
+                CONF_BLOCKTYPES: validBlockTypes,
+            }
         elif self.geometry == "1D cylinder":
-            defaults = dict(
-                criticalBuckling=False,
-                mergeIntoClad=[],
-                blockRepresentation=blockRepresentation,
-                numExternalRings=1,
-                validBlockTypes=validBlockTypes,
-            )
+            defaults = {
+                CONF_GEOM: "1D cylinder",
+                CONF_DRIVER: "",
+                CONF_MERGE_INTO_CLAD: ["gap"],
```

Comment:
this should probably be in a SFR design specific plugin. could be a framework plugin but doesn't seem right here. right?

Reply:
I am not looking to refactor this out right now, just trying to clarify which are the actual expected defaults rather than the ones that were just put here before.

Reply:
It seems that this implementation of the crossSectionSettings is pretty much all design specific, but without this then the behavior of the `CrossSectionGroupManager` is not well undefined. Other than `fileLocation`, `blockRepresentation`, and `validBlockTypes` probably all other options should be removed. `geometry` probably also makes sense because it feeds into the `latticePhysicsInterface` in `getWriters`.